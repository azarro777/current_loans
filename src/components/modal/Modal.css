.modal {
	position: absolute;
	width: 1440px;
	height: 1024px;
	left: 0px;
	top: 0px;
	background: rgba(26, 26, 26, 0.8);
	transform: scale(0);
}

.modal.active {
	transform: scale(1);
}

.modal-content {
	width: 627px;
	height: 499px;
	background: #FFFFFF;
	border-radius: 5px;
	position: absolute;
	left: 407px;
	top: 263px;
}

.modal-close__button {
	border: none;
	background-color: transparent;
	position: absolute;
	width: 32px;
	height: 32px;
	left: 563px;
	top: 32px;
	padding: 0;
}

.modal-content__block {
	position: absolute;
	width: 499px;
	height: 371px;
	left: 64px;
	top: 64px;
}

.modal-title {
	font-family: 'Inter';
	font-style: normal;
	font-weight: 600;
	font-size: 35px;
	line-height: 42px;
	color: #303247;
	margin: 0;
	padding: 0;
}

.modal-title__clicked {
	font-family: 'Inter';
	font-style: normal;
	font-weight: 400;
	font-size: 20px;
	line-height: 24px;
	margin-top: 30px;
	padding: 0;
	color: #303247;
}

.modal-title__amount {
	font-family: 'Inter';
	font-style: normal;
	font-weight: 400;
	font-size: 20px;
	line-height: 24px;
	margin: 32px 0 0 0;
	padding: 0;
	color: #303247;
}

.modal-title__ends {
	overflow: hidden;
	font-family: 'Inter';
	font-style: normal;
	font-weight: 400;
	font-size: 20px;
	line-height: 24px;
	margin-top: 10px;
	padding: 0;
	color: #303247;
}

.modal-title__investment {
	font-family: 'Inter';
	font-style: normal;
	font-weight: 600;
	font-size: 30px;
	line-height: 36px;
	margin-top: 65px;
	padding: 0;
	color: #303247;
}

.modal-input__block {
	display: flex;
	justify-content: space-between;
}

.modal-title__input {
	display: flex;
	flex-direction: row;
	justify-content: flex-end;
	align-items: center;
	padding: 16px 32px;
	gap: 10px;
	width: 264px;
	height: 24px;
	background: #FFFFFF;
	border: 1px solid rgba(48, 50, 71, 0.1);
	border-radius: 5px;
	margin: 0;
	text-align: right;
	font-family: 'Inter';
font-style: normal;
font-weight: 400;
font-size: 20px;
line-height: 24px;

color: #303247;
}

.modal-title__button {
	
	padding: 16px 32px;
	/* gap: 10px; */

/* position: absolute; */
width: 137px;
height: 56px;
/* left: 426px;
top: 379px; */
border: none;
background: #F7E308;
border-radius: 5px;
font-family: 'Inter';
font-style: normal;
font-weight: 500;
font-size: 20px;
line-height: 24px;
text-transform: uppercase;

color: #303247;
}