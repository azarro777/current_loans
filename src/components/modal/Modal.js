import { useState } from 'react';
import closeImg from './close.svg';
import { setAvailableAmount, setInvest, setModal } from '../../reducers/appReducer';
import { useDispatch, useSelector } from 'react-redux';
import { convertSeconds } from '../../utils/convertSeconds';
import './Modal.css';

export const Modal = () => {
	const dispatch = useDispatch();
	const isModalActive = useSelector(state => state.app.isModalActive);
	const modalData = useSelector(state => state.app.modalData);
	const {available, term_remaining} = modalData;
	
	const [modalValue, setModalValue] = useState('');

	const investHandler = () => {
		dispatch(setAvailableAmount(Number(modalValue)));
		setModalValue('');
	};

	const closeHandler = () => {
		dispatch(setModal(false));
		dispatch(setInvest(modalData));
	};

	return (
	<div className={isModalActive ? "modal active" : "modal"}>
		<div className="modal-content">
			<button className="modal-close__button" onClick={() => closeHandler()}>
				<img src={closeImg} alt="close"/>
			</button>
			<div className="modal-content__block">
				<h1 className="modal-title">Invest in Loan</h1>
				<h3 className="modal-title__clicked">Loan title youâ€™ve clicked</h3>
				<h3 className="modal-title__amount">Amount available: ${available}</h3>
				<h3 className="modal-title__ends">Loan ends in: {convertSeconds(term_remaining)}</h3>
				<h2 className="modal-title__investment">Investment amount</h2>
				<div className="modal-input__block">
					<input 
						type="number"
						value={modalValue} 
						onChange={(event) => setModalValue(event.target.value)} 
						className="modal-title__input"
					/>
					<button onClick={() => investHandler()} className="card-button">INVEST</button>
				</div>
			</div>
		</div>
	</div>
);
};