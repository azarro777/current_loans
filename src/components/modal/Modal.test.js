import React from 'react';
import { Modal } from './Modal';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';

describe("on mount", () => {
  it("dispatch search action to store", () => {
		const mockStore = configureMockStore([thunk])([]);;
		const store = mockStore;
		shallow(<Provider store={store}><Modal/></Provider>)
    const actions = store.getActions();
    expect(actions).toEqual([]);
  });
	it('Modal have rendered correctly', () => {
		const mockStore = configureMockStore([thunk])([]);;
		const store = mockStore;
		const modal = shallow(<Provider store={store}><Modal/></Provider>);
		expect(modal).toMatchSnapshot();
	});
});