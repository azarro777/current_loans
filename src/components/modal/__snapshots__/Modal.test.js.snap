// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on mount Modal have rendered correctly 1`] = `ShallowWrapper {}`;
